### PR TITLE
Homework 7

### PR DESCRIPTION
**Pull request by kalimullina**

### CODE REVIEW

File: src/main/java/com/epam/tc/hw7/entities/Data.java [hunk 1]
```diff
@@ -4,13 +4,13 @@
 
 public class Data {
```
Comment (pnatashap) [inline src/main/java/com/epam/tc/hw7/entities/Data.java]: Data is a very generic name, should be specified for real content

File: src/main/java/com/epam/tc/hw7/pages/JdiMetalsAndColorsPage.java [hunk 2]
```diff
@@ -3,57 +3,117 @@
...
+        for (String vegetable : vegetables) {
+            vegetablesSettings.select(vegetable);
+        }
+    }
+
+    public void submitFormMetalsColors() {
+        submitButton.click();
+    }
+
+    public void checkResultLog(Data dataFromJson) {
```
Comment (pnatashap) [inline src/main/java/com/epam/tc/hw7/pages/JdiMetalsAndColorsPage.java]: Very difficult to understand this function. Why you cann't just read text from Page and compare with the string from the data?

File: src/test/java/com/epam/tc/hw7/JdiTest.java [hunk 3]
```diff
@@ -43,5 +33,22 @@ public void loginTest() {
...
+        JdiSite.metalsAndColorsPage.chooseVegetables(dataFromJson.getVegetables());
+
+        //Submit form Metals & Colors
+        JdiSite.metalsAndColorsPage.submitFormMetalsColors();
+
+        //Result sections should contains data from json
+        JdiSite.metalsAndColorsPage.checkResultLog(dataFromJson);
+
+        //Logout
+        JdiSite.userPage.logoutFromUserAccount();
```
Comment (pnatashap) [inline src/test/java/com/epam/tc/hw7/JdiTest.java]: It's a bad practice to login and logout for each test run. Should login only once

**Review by pnatashap — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/tc/hw7/entities/Data.java (hunk 1 above)
Comment (kalimullina) [reply src/main/java/com/epam/tc/hw7/entities/Data.java]: changed

File: src/main/java/com/epam/tc/hw7/pages/JdiMetalsAndColorsPage.java (hunk 2 above)
Comment (kalimullina) [reply src/main/java/com/epam/tc/hw7/pages/JdiMetalsAndColorsPage.java]: changed

File: src/test/java/com/epam/tc/hw7/JdiTest.java (hunk 3 above)
Comment (kalimullina) [reply src/test/java/com/epam/tc/hw7/JdiTest.java]: changed

File: pom.xml [hunk 4]
```diff
@@ -206,15 +206,6 @@
             <version>1.3.13</version>
         </dependency>
 
-        <!--For json parsing-->
-        <dependency>
-            <groupId>com.google.code.gson</groupId>
```
Comment (pnatashap) [inline pom.xml:211]: Now I have compilation error due to missing dependency

File: src/main/java/com/epam/tc/hw7/pages/JdiMetalsAndColorsPage.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        submitButton.click();
+    }
+
+    public void checkResultLog(DataFromJson dataFromJson) {
+
+        int summaryActualResult = dataFromJson.getSummary().get(0) + dataFromJson.getSummary().get(1);
+        summaryResult.shouldBe().text(Matchers.equalTo("Summary: " + summaryActualResult));
+
+        for (String element : dataFromJson.getElements()) {
+            elementsResult.shouldBe().text(Matchers.containsString(element));
```
Comment (pnatashap) [inline src/main/java/com/epam/tc/hw7/pages/JdiMetalsAndColorsPage.java]: containsString is a very dangerous matcher. Firewall contains Fire, but it's not correct. Still very difficult to understand the logic of this verification step.

File: src/main/java/com/epam/tc/hw7/pages/JdiUserPage.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+import com.epam.jdi.light.elements.complex.Menu;
+import com.epam.jdi.light.elements.composite.WebPage;
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.UI;
+import com.epam.jdi.light.ui.html.elements.common.Button;
+import com.epam.tc.hw7.entities.MenuOptions;
+
+public class JdiUserPage extends WebPage {
+
+    @UI("header ul[class='uui-navigation nav navbar-nav m-l8']")
```
Comment (pnatashap) [inline src/main/java/com/epam/tc/hw7/pages/JdiUserPage.java]: Very dangerous locator, you don't need all these classes in the list

File: src/test/java/com/epam/tc/hw7/JdiTest.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        //Login on JDI site as User
+        JdiSite.login(User.ROMAN);
+        JdiSite.homePage.checkLoggedin(User.ROMAN);
+    }
+
+    @Test(dataProviderClass = DataProviderJdi.class, dataProvider = "jsonData")
+    public void loginTest(DataFromJson dataFromJson) {
+
+        //Open Metals & Colors page by Header menu
+        JdiSite.userPage.openPageFromHeaderMenu(MenuOptions.METALS_AND_COLORS);
```
Comment (pnatashap) [inline src/test/java/com/epam/tc/hw7/JdiTest.java:41]: Before each test you should be sure, that you are still logged in

**Review by pnatashap — CHANGES_REQUESTED** [review]
Comments are in a code

**Review by pnatashap — APPROVED** [review]

**Review by pnatashap — APPROVED** [review]

**Pull request merged**